doc/implementation_comparison.md

```diff
@@ -0,0 +1,181 @@
+# grpc-dotnet and Grpc.Core comparison
+
+This document summarizes the differences between the two available implementations of gRPC in C#.
+
+## Where both implementations are the same
+
+- 100% wire compatible and interoperable with each other and with other gRPC implementations
+- same API for invoking and handling RPC calls
+- basic functionality (streaming, metadata, deadline, cancellation, ...)
+- using same codegen tooling and MSBuild Integration (Grpc.Tools)
+- differences in setting up server and client are documented here: https://docs.microsoft.com/en-us/aspnet/core/grpc/migration?view=aspnetcore-3.0
+
+## Criteria for choosing between grpc-dotnet and gRPC C#
+
+One might choose one or the other implementation mostly for one of these reasons
+
+- avoid use of native code
+- ability to use .NET Core 3 and ASP.NET Core 3 (it's a brand new stack so not everyone will be able to use immediately)
+- want seamless integration with ASP.NET Core 3, dependency injection etc.
+- features available (see breakdown)
+- maturity level
+- performance (TODO: add data)
+
+## Comparison of supported features 
+
+Beyond the basic RPC functionality, there are a lot of gRPC features that may or may not be supported. The summary
+of supported features in both implementation is available in this section.
+
+### Service config
+
+Service config is currently only supported by Grpc.Core. Right now the feature is not that useful 
+because support for service config encoded in DNS records hasn't been enabled yet by default.
+
+Also see:
+
+- https://github.com/grpc/grpc/blob/master/doc/service_config.md
+- https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md 
+
+### Tracing
+
+grpc-dotnet has tracing bindings that come from ASP.NET Core, and it works with OpenTelemetry SDK, OpenCensus support is TBD.
+Grpc.Core: currently not supported.
+
+### Interceptors
+
+Both implementations support client and server interceptors from Grpc.Core.Interceptors namespace. Interceptors operate at post-deserialization and pre-serialization level (no access to binary payloads).
+
+In addition to gRPC-aware interceptors, grpc-dotnet also allows interception at a HTTP/2 level because under ASP.NET core, all RPCs are also HTTP/2 handlers. 
```

Comment:
grpc-dotnet supports middleware on the server, and [HttpMessageHandlers](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpmessagehandler) on the client

Reply:
Is that meant to be added or is it a rephrase of what I mentioned as "interception at a HTTP/2 level"? Feel free to suggest an edit.